internal/transaction/transaction.service.go

```diff
@@ -0,0 +1,136 @@
+package transaction
+
+import (
+	"Pelter_backend/internal/dto"
+	"Pelter_backend/internal/entity"
+	"Pelter_backend/internal/pkg/jwt"
+	"strconv"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+type (
+	transactionService struct {
+		transactionUsecase TransactionUsecase
+	}
+	TransactionService interface {
+		CreateTransaction(ctx *fiber.Ctx) error
+		GetTransactions(ctx *fiber.Ctx) error
+		GetTransactionByID(ctx *fiber.Ctx) error
+		GetTransactionsByUserID(ctx *fiber.Ctx) error
+	}
+)
+
+func NewTransactionService(transactionUsecase TransactionUsecase) TransactionService {
+	return &transactionService{
+		transactionUsecase: transactionUsecase,
+	}
+}
+
+func (s *transactionService) CreateTransaction(ctx *fiber.Ctx) error {
+	idParam := ctx.Params("id")
+	id, err := strconv.ParseUint(idParam, 10, 64)
+	if err != nil {
+		return ctx.Status(fiber.StatusBadRequest).JSON(dto.HttpResponse{
+			Error:   "Invalid ID",
+			Success: false,
+		})
+	}
+	userId, err := jwt.GetIDFromToken(ctx.Cookies("access_token"))
+	if err != nil {
+		return ctx.Status(fiber.StatusInternalServerError).JSON(dto.HttpResponse{
+			Error:   "Cannot get UserID from access_token context: " + err.Error(),
+			Success: false,
+		})
+	}
+	txn := entity.Transaction{
+		ProductID: uint(id),
+		BuyerID:   userId,
+	}
+	newTxn, err := s.transactionUsecase.CreateTransaction(ctx.UserContext(), &txn)
+	if err != nil {
+		switch err.Error() {
```

Comment:
Verify the use of HTTP status codes I don't think Conflict or Not Found are being used correctly

Reply:
f1c7cc53cb4367ec5429ee274dcdb7185a11aac6 would solve this